paystell-frontend/src/components/transaction/transaction-history-dashboard.tsx

```
@@ -0,0 +1,261 @@
+      } else if (filters.status === "failed") {
+        filteredRecords = filteredRecords.filter((tx) => !tx.successful)
+      }
+
+      // Apply search filter
+      if (filters.searchQuery) {
+        const query = filters.searchQuery.toLowerCase()
+        filteredRecords = filteredRecords.filter(
+          (tx) =>
+            tx.id.toLowerCase().includes(query) ||
+            tx.source_account.toLowerCase().includes(query) ||
+            (tx.memo && tx.memo.toLowerCase().includes(query)),
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)